php/php.dbgp/src/org/netbeans/modules/php/dbgp/DebugSession.java

```diff
@@ -124,6 +125,14 @@ public void run() {
                 } catch (Throwable e) {
                     log(e, Level.SEVERE);
                 }
+                if (canceled) {
+                    synchronized (commands) {
```

Comment:
The change looks good to me. However, the threading in this class is not documented and is not clear to me - I can see no synchronization of `commands` in `preprocess()` method, for example.

Reply:
You are right!

Reply:
Fixed only it: https://github.com/apache/netbeans/compare/79f57ddf2076828384d9b9a10ab082e44cb1c594..582c2c57d1e053833e7dafc7aedf34abc3b70187